app/mm-redux/reducers/entities/groups.ts

```diff
@@ -196,7 +196,22 @@ function groups(state: Dictionary<Group> = {}, action: GenericAction) {
         }
         return nextState;
     }
+    case GroupTypes.RECEIVED_ALL_GROUPS_ASSOCIATED_TO_CHANNELS_IN_TEAM: {
+        const nextState = {...state};
+        const {groupsByChannelId} = action.data;
+
+        for (const channelID of Object.keys(groupsByChannelId)) {
```

Comment:
You are executing this reducer even if the result of the request was an empty array, and thus updating the state withou need, same with the others

Reply:
Done! Now I check if data is empty before dispatching any actions to ensure we don't perform an action on empty data.